File: leetcode-9/python/leetcode-9.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :param x: int
        :return: bool
        """
        if x < 0:
            return False
        if x < 10:
            return True
        str_num = str(x)
        str_reverse = str_num[::-1]
        print(str_reverse)
        return str_reverse == str_num

    def isPalindrome2(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        if x < 10:
            return True
        temp_x = x
        reverse_x = 0
        while temp_x:
            reverse_x = reverse_x * 10 + (temp_x % 10)
            temp_x = temp_x // 10       #  整除
        return reverse_x == x

s = Solution()

s.isPalindrome(222)

File: leetcode-9/javascript/leetcode-9.js
/**
 * @param {number} x
 * @return {boolean}
 */
function isPalindrome(x) {
  if (x < 0) {
    return false;
  }
  if (x < 10) {
    return true;
  }
  let tempx = x;
  let reverse = 0;
  while (tempx) {
    reverse = reverse * 10 + tempx % 10;
    tempx = Math.floor(tempx / 10);
  }
  console.log(reverse);
  return x === reverse;
}

function isPalindrome2(x) {
  if (x < 0) {
    return false;
  }
  if (x < 10) {
    return true;
  }
  let str_num = x.toString();
  for (let i = 0; i < str_num.length/2; ++i) {
    if (str_num[i] != str_num[str_num.length - 1 - i]) {
      return false;
    }
  }
  return true;
}

console.log(isPalindrome2(-121));